### ISSUE TITLE
Trouble displaying an invited user's username once it's persisted in DB

### ISSUE THREAD

**Issue by ericacomfort** [ga-wdi-exercises/project4#60]
Currently, when a user creates an event and is on their event trying to add users, the proper username shows up properly when it's added on the front end, but seems to be formatted differently once it's stored in the back end.

I'm confused about what I need to change (I guess on the back end) that will display the username on an ng-repeat.

Here is method on the form: https://github.com/andersenek/gracious.guest/blob/master/gracious_guest/views/index.ejs#L312

Which hits the `addUser` method in the events controller here:
https://github.com/andersenek/gracious.guest/blob/master/gracious_guest/public/javascripts/angularApp.js#L297-L308

That hits the `addUser` method on the events in the factory: https://github.com/andersenek/gracious.guest/blob/master/gracious_guest/public/javascripts/angularApp.js#L187-L195

The back end route is:
https://github.com/andersenek/gracious.guest/blob/master/gracious_guest/routes/events.js#L183-L212

So long story short, my front end template using `ng-repeat` properly displays `user.username` before a refresh (disappears on refresh)
And if I display `user` the front end template displays `{"__v":0,"username":"anewusertoaddtothisevent","_id":"55fc42bbf8c9d6fb3c50a8e6","event":[]}`
and on refresh shows just the id.

For my app you'll need to run:
`npm install`
`mongod`
`npm start`

Any help to get this to display the username when it's called on the front end and back end would be greatly appreciated!

**Comment by RobertAKARobin** [comment]
Looking now. :+1: for having the best-written issues!

**Comment by RobertAKARobin** [comment]
What the eff.

**Comment by ericacomfort** [comment]
It's so strange, I was hoping it'd be something small that I missed :(

**Comment by RobertAKARobin** [comment]
Found it. You need to populate users:

``` js
router.get('/events/:event', function(req, res) {
  req.event.populate('comments', function(err, event) { // Populate event with comments
    event.populate('users', function(err, output){
      res.json(output)
    })
  });
});
```

**Comment by ericacomfort** [comment]
Oh my goodness! Thank you so much for your help this week!

**Comment by RobertAKARobin** [comment]
I like this syntax better:

``` js
  Event
  .findById(req.params.event)
  .populate('comments')
  .populate('users')
  .exec(function(err, data){
      res.json(data)
  });
```

We won't talk about how long this took me to figure out.
